utils/swift_build_support/swift_build_support/products/swiftpm.py

```diff
@@ -2,16 +2,72 @@
 #
 # This source file is part of the Swift.org open source project
 #
-# Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+# Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
 # Licensed under Apache License v2.0 with Runtime Library Exception
 #
 # See https://swift.org/LICENSE.txt for license information
 # See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 #
 # ----------------------------------------------------------------------------
 
-from . import product
+import os
+import platform
 
+from . import product
+from .. import shell
+from .. import targets
 
 class SwiftPM(product.Product):
-    pass
+    @classmethod
+    def product_source_name(cls):
+        return "swiftpm"
+
+    @classmethod
+    def is_build_script_impl_product(cls):
+        return False
+
+    def build(self, host_target):
+        run_build_script_helper('build', host_target, self, self.args)
+
+    def test(self, host_target):
+        run_build_script_helper('test', host_target, self, self.args)
```

Comment:
You should not run the tests unless your test flag is passed. See 
https://github.com/apple/swift/pull/27580/files#diff-bac1aa9ca5d00ef87144461e6bc764edR77

(this should actually be changed that the driver takes care of checking the flag)

Reply:
huh, ok.